rskj-core/src/main/java/co/rsk/rpc/modules/trace/TraceModuleImpl.java

```diff
@@ -254,7 +249,7 @@ private List<TransactionTrace> buildBlockTraces(Block block, TraceFilterRequest
 
                 if (programTrace == null) {
                     blockTraces.clear();
-                    return null;
+                    return Collections.emptyList();
```

Comment:
It was there before, but why clearing the collection if you are just returning a different thing? Just either clear it and `break` or `return Collections.emptyList();`

I would also probably take the chance refactor this method (just a bit) to early exit with an empty collection in line 220 `if (block == null || block.getNumber() == 0)` in order to reduce indentation and ease readability.

Reply:
The idea for this PR was to solve only this sonar finding, we can add all those other refactoring changes in another PR for the sake of clarity.